src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py

```diff
@@ -479,33 +479,47 @@ def get_frame_by_row_pos_slice_frame(
     frame = internal_frame.ensure_row_position_column()
     row_pos_col = col(frame.row_position_snowflake_quoted_identifier)
 
-    # create a count column which will be used to check step
-    frame = frame.append_column("count", pandas_lit(1) + max_(row_pos_col).over())
-    count_col = col(frame.data_column_snowflake_quoted_identifiers[-1])
+    def get_count_col() -> Column:
+        nonlocal frame
```

Comment:
actually, there is one part i am little bit confused here, what is the purpose to wrap this into a function? or we can just do the following like what we had before
```
frame = frame.ensure_row_count_column()
count_col = col(frame.row_count_snowflake_quoted_identifier)
```
was this fixing a bug? or for other purpose?

Reply:
The purpose is to avoid creating row count column if it is not needed. Creating row count column will cause the query to scan the whole table.

Reply:
actually, never mind, i think i got what you mean here. so you only want to call this when needed